quesma/clickhouse/clickhouse.go

```diff
@@ -482,8 +496,38 @@ func generateNonSchemaFieldsString(attrsMap map[string][]interface{}) (string, e
 	return nonSchemaStr, nil
 }
 
+// This function implements heuristic for deciding if we should add new columns
+func (lm *LogManager) shouldAlterColumns(table *Table, attrsMap map[string][]interface{}) (bool, []int) {
```

Comment:
Ok, I think the below model should work:
1. Below 100, always add.
2. Till 1000 use heuristic.
3. Don't add above that number.

Reply:
As discussed added upper limit for now. We can simplify this in next PR